python/lsst/pipe/tasks/transformMeasurement.py

```diff
@@ -152,38 +184,172 @@ class RunTransformTask(pipeBase.CmdLineTask):
+
+    # Subclasses should provide appropriate definitions for the attributes named below.
+    # Properties can be used if apprpriate.
+    #
+    # Standard CmdLineTask attributes:
+    _DefaultName = None
+
+    # Boolean; True if the measurement operation was forced, otherwise False.
+    wasForced = None
+
+    # Butler dataset type of the source type to be transformed ("src", "forced_src", etc):
+    sourceType = None
+
+    # Butler dataset type of the calibration exposure to use when transforming ("calexp", etc):
+    calexpType = None
+
+    @property
+    def inputSchemaType(self):
+        """!
+        The Butler dataset type for the schema of the input source catalog.
+
+        By default, we append `_schema` to the input source type. Subclasses may customise
+        if required.
+        """
+        return self.sourceType + "_schema"
+
+    @property
+    def outputType(self):
+        """!
+        The Butler dataset type for the schema of the output catalog.
+
+        By default, we prepend `transformed_` to the input source type. Subclasses may
+        customise if required.
+        """
+        return 'transformed_' + self.sourceType
+
+    @property
+    def measurementConfig(self):
+        """!
+        The configuration of the measurement operation used to generate the input catalog.
+
+        By default we look for `measurement` under the root configuration of the
+        generating task. Subclasses may customise this (e.g. to `calibration.measurement`)
+        if required.
+        """
+        return self.butler.get(self.config.inputConfigType).measurement.value
 
     def __init__(self, *args, **kwargs):
         pipeBase.CmdLineTask.__init__(self, *args, config=kwargs['config'], log=kwargs['log'])
-        if self.config.measType == "SingleFrame":
-            pluginRegistry = measBase.sfm.SingleFramePlugin.registry
-        elif self.config.measType == "Forced":
+        if self.wasForced:
             pluginRegistry = measBase.forcedMeasurement.ForcedPlugin.registry
+        else:
+            pluginRegistry = measBase.sfm.SingleFramePlugin.registry
+
+        self.butler = kwargs['butler']
+
         self.makeSubtask('transform', pluginRegistry=pluginRegistry,
-                         measConfig=kwargs['butler'].get(self.config.measConfig).measurement.value,
-                         inputSchema=kwargs['butler'].get("src_schema").schema)
+                         measConfig=self.measurementConfig,
+                         inputSchema=self.butler.get(self.inputSchemaType).schema,
+                         outputType=self.outputType)
 
     @pipeBase.timeMethod
     def run(self, dataRef):
         """!Transform the source catalog referred to by dataRef.
 
-        The result is both returned and written as dataset type "transformedSrc"
-        to the provided dataRef.
+        The result is both returned and written as dataset type "transformed" + the input
+        source type to the provided dataRef.
```

Comment:
It's actually `"transformed_" + the input source type` (note underscore), and you probably want to say "source dataset" rather than "source type".